backstory/src/main/java/com/google/sps/servlets/data/BlobstoreManager.java

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets.data;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.sps.servlets.data.BlobstoreServiceConstantFields;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Wrapper class for the blobstore service and all related operations.
+ */
+public class BlobstoreManager {
+  private BlobstoreService blobstoreService;
+  private BlobstoreServiceConstantFields blobstoreServiceConstantFields;
+  private BlobInfoFactory blobInfoFactory;
+
+  /**
+   * Creates a blobstore manager object by calling the blobstore service factory
+   */
+  public BlobstoreManager() throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    this.blobstoreService = blobstoreService;
+    this.blobstoreServiceConstantFields = new BlobstoreServiceConstantFields(blobstoreService);
+    this.blobInfoFactory = new BlobInfoFactory();
+  }
+
+  /**
+   * Injection code for testing.
+   * Creates a blobstore manager using a mock blobstoreService.
+   */
+  public BlobstoreManager(BlobstoreService blobstoreService,
+      BlobstoreServiceConstantFields blobstoreServiceConstantFields,
+      BlobInfoFactory blobInfoFactory) {
+    this.blobstoreService = blobstoreService;
+    this.blobstoreServiceConstantFields = blobstoreServiceConstantFields;
+    this.blobInfoFactory = blobInfoFactory;
+  }
+
+  /**
+   * Given a request and the form input element name get the image uploaded in the form as
+   * a Blob Key in String form. This key is used to serve the picture back to the front-end.
+   * @param request the HTTP request sent from the front-end form.
```

Comment:
Nit: I think according to the style guide there should be a line in between the description and the tags.